include/deal.II/grid/grid_generator.h

```diff
@@ -388,6 +433,41 @@ namespace GridGenerator
                   const double        half_length = 1.0);
 
   /**
+  * \brief A center cell with stacks of cell protruding from each surface.
+  *
+  * Each of the square mesh cells is Cartesian and has size one in each
+  * coordinate direction. The center of cell number zero is the
+  * origin.
+  *
+  * @param tria A Triangulation object which has to be empty.
+  *
+  * @param sizes A vector of integers of dimension
+  * GeometryInfo<dim>::faces_per_cell with the following meaning: the
+  * legs of the cross are stacked on the faces of the center cell,
+  * in the usual order of deal.II cells, namely first $-x$, then $x$,
+  * then $-y$ and so on. The corresponding entries in
+  * <code>sizes</code> name the number of cells stacked on this
+  * face. All numbers may be zero, thus L- and T-shaped domains are
+  * specializations of this domain.
+  *
+  * @param colorize If colorization is chosen, then the material id of a cells
+  * corresponds to the leg it is in. The id of the center cell is zero,
+  * and then the legs are numbered starting at one.
+  *
+  * Examples in two and three dimensions are
+  *
+  * @image hyper_cross_2d.png
+  * @image hyper_cross_3d.png
+  *
+  * @author Guido Kanschat
+  * @date 2015
+  */
+  template <int dim, int spacedim>
+  void hyper_cross(Triangulation<dim, spacedim> &tria,
+                   const std::vector<unsigned int> &sizes,
+                   const bool colorize = false);
```

Comment:
I think if you use the word "colorize" in a context that allows it to be confused with how it is used in all of the other places in this namespace, then it would be useful to either explicitly point this out in the documentation (I missed this sentence, and would have probably thought that you mistyped boundary as material id), or to use a different word.